default.nix

```diff
@@ -1,19 +1,11 @@
 let
   localLib = import ./lib.nix;
-  jemallocOverlay = self: super: {
-    # jemalloc has a bug that caused cardano-sl-db to fail to link (via
-    # rocksdb, which can use jemalloc).
-    # https://github.com/jemalloc/jemalloc/issues/937
-    # Using jemalloc 510 with the --disable-initial-exec-tls flag seems to
-    # fix it.
-    jemalloc = self.callPackage ./nix/jemalloc/jemalloc510.nix {};
-  };
 in
 { system ? builtins.currentSystem
 , config ? {}
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; overlays = [ (import ./nix/overlays/jemalloc.nix) ]; })
 , gitrev ? localLib.commitIdFromGitRepo ./.git
 , buildId ? null
-, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; overlays = [ jemallocOverlay ]; })
 , forceDontCheck ? false
```

Comment:
changes in this area are likely to conflict with https://github.com/input-output-hk/cardano-sl/pull/3721, try holding this PR off until after that has merged maybe

Reply:
This seems like a manageable merge conflict; I'm also not certain when #3721 will be merged, so I'm fine with this to go ahead first.